force-app/main/default/classes/CBCreateSubscriptionTest.cls

```diff
@@ -1,397 +1,473 @@
+    {
+     	cbs.plans='TieredPlan';
+        Integer initialQuantity=11;        
+        cbs.pquan=initialQuantity;
+        cbs.CbPriceBookEntry();
+          System.debug(' TieredPlanExpected: Price ' + cbs.planPrice + ' displayIcon ' + cbs.displayIcon 
+                     + ' TierData ' + cbs.tierdata + ' showUnit ' + cbs.showUnit);  
+        String expectedPrice='1750.00'; //data populated in testsetup        
+        System.assertEquals(expectedprice,cbs.planPrice);
+        System.assert(cbs.displayIcon);
+        System.assert(cbs.tierdata != null);           
+        zeroQuantityCheck(cbs);
         
-         //chargebeeapps__CB_Pricing_Tiers__c[] cbtiers = [select Id,Name,chargebeeapps__ExternalId__c from  chargebeeapps__CB_Pricing_Tiers__c];
-		 Test.startTest();
-         PageReference pageRef = Page.Create_Subscription;
-         test.setCurrentPageReference(pageRef);
-         pageRef.getParameters().put('opp_id',oppid);
-         //pageRef.getParameters().put('opp_id',null);
-         pageRef.getParameters().put('con_id',coid);
-         pageRef.getParameters().put('domain','cbtest1');
-         Test.setMock(HttpCalloutMock.class, new MockSubGenerator());         
-         CBCreateSubscription cbn = new CBCreateSubscription();
-         cbn.selectedSites='cbtest1';
-         cbn.plans='Testplan1';         
-         cbn.sameBillAddress = false;
-         cbn.planPrice = '100';          
-         cbn.sDate = system.today();
-         cbn.invNow=true;
-         cbn.stdate=true;        
-         cbn.preview();
-         cbn.populate();
-         cbn.selecttab();
-         cbn.selecttab1();                 
-         cbn.cancel_page();
-         Integer callouts = Limits.getCallouts();
-         Test.stopTest();
-         system.assertequals(2,callouts);
-     }
-
-    @IsTest public static void testCheckout1() 
-     {
-         Account acc = [select Id,Name from Account where Name='SfTest1' LIMIT 1];
-         system.debug('acc id'+acc.Id);
-         chargebeeapps__CB_Sites__c sites = [select name from chargebeeapps__CB_Sites__c  where name='cbtest1'];
-         Contact con = [select Id,LastName from Contact where LastName='test1' LIMIT 1];
-         chargebeeapps__Contact_Associate_Site__c testCon =[select chargebeeapps__External_Id__c,chargebeeapps__Contact__c,chargebeeapps__CB_Site__c from chargebeeapps__Contact_Associate_Site__c where chargebeeapps__Contact__c=:con.Id];
-    	 system.debug('testCon'+testCon);
-         Opportunity opt = [select Id,Name,StageName,CloseDate from Opportunity where Name='test opp2' LIMIT 1];
-         string oppid = opt.Id;
-         string coid = con.Id;
-         Product2[] prd = [select Name,ExternalId from Product2];
-         system.debug('prd--'+prd);   
-         PricebookEntry[] selctedprod = [select Id,Product2.Name,Product2.ExternalId from PricebookEntry];
-         system.debug('selctedprod--'+selctedprod);
-         //chargebeeapps__CB_Pricing_Tiers__c[] cbtiers;
+    }
```

Comment:
1) site change logic
2) opportunity pricebook is the same as the site pricebook
3) opportunity pricebook is different from site pricebook
4) Addon addition. Addon removsl, Addon change, Addon qty change, Addon price, tiers, override verification. 
5) Saving the subscription - validate all the inputs - plans, addons, address, invoice options, preview page UI elements need to be verified, confirm - verify the callouts, redirect url.

Reply:
Addon Section and Saving Subscription is done as mentioned above Validate Invoice Options,Callout, Redirection is done. 
NOT DONE : Address validation is not done, because the source directly sends the values irrespective of Empty fields or not. Hence ignored the Address Name validations.

Reply:
Site change is Handled by loading different products/addons for different sites. The size for the Plans/Addons are compared after change.